lib/streamlit/commands/execution_control.py

```diff
@@ -54,22 +59,71 @@ def stop() -> NoReturn:  # type: ignore[misc]
         st.empty()
 
 
+def _new_fragment_id_queue(
+    ctx: ScriptRunContext,
+    scope: Literal["app", "fragment"],
+) -> list[str]:
+    if scope == "app":
+        return []
+
+    else:  # scope == "fragment"
+        curr_queue = (
+            ctx.script_requests.fragment_id_queue if ctx.script_requests else []
+        )
+
+        # TODO(vdonato): Figure out if this behavior is reasonable. We'll want to
+        # discuss this with product. The tricky case is that users may run into this
+        # if st.rerun(scope="fragment") is called within a fragment on the first script
+        # run.
+        # Other options include:
+        #   * Rerunning just this fragment even if this was triggered in a full run
+        #     (implementation for this may be tricky)
+        #   * Just rerunning the full script
+        # After more thought, strongly prefer either erroring or doing a full script run
+        # because weird stuff can happen if we try to rerun just a fragment before a
```

Comment:
I feel we should talk about this "weird stuff can happen" and fill it with some color. Intuitively its not clear to me why it should be a problem because for the fragment to run in the first place it can only reference stuff that has already been executed. So, everything that comes after the fragment code should be irrelevant to the code function, shouldn't it?

Reply:
Rewrote this comment to describe the weirdness with allowing `st.rerun(scope="fragment")` from within a fragment but during a full script run.